apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ir/Argument.kt

```diff
@@ -1,5 +1,8 @@
 package com.apollographql.apollo.compiler.ir
 
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
 data class Argument(
```

Comment:
One main side effect of this pull request is that the IR class have been made Json serializable. I think that's ok but let me know if there's another way to look at this.

Reply:
Could you please elaborate more on this, why it's serializable?

Reply:
The IR is now used as a task input/output so Gradle will need it be serialized for `UP-TO-DATE` checks. 

The `generateOperationIds` tasks needs access to the `sourceWithFragments` and most of that is the result of parsing the `.graphql` files. 

The other option is to keep everything bundled in one single task and keep relying on the `OperationIdGenerator.version` for invalidation, which feels less "Gradle-ly".

Something else to consider is multi-modules. I've been thinking about that lately and the IR would be a convenient point to inject fragments and types from another module.

Reply:
Serialization is now removed